engine/language_client_codegen/src/openapi.rs

```diff
@@ -326,22 +326,26 @@ impl<'ir> TryFrom<(&'ir IntermediateRepr, &'_ crate::GeneratorArgs)> for OpenApi
     type Error = anyhow::Error;
 
     fn try_from((ir, _): (&'ir IntermediateRepr, &'_ crate::GeneratorArgs)) -> Result<Self> {
-        Ok(Self {
-            paths: ir
-                .walk_functions()
-                .map(|f| {
-                    f.try_into().context(format!(
-                        "Failed to convert BAML function {} to OpenAPI method",
-                        f.item.elem.name()
-                    ))
-                })
-                .collect::<Result<_>>()?,
-            schemas: vec![]
-                .into_iter()
-                .chain(ir.walk_enums().map(|e| Ok((e.name(), e.try_into()?))))
-                .chain(ir.walk_classes().map(|c| Ok((c.name(), c.try_into()?))))
-                .collect::<Result<_>>()?,
-        })
+        let mut paths: Vec<OpenApiMethodDef> = ir
+            .walk_functions()
+            .map(|f| {
+                f.try_into().context(format!(
+                    "Failed to convert BAML function {} to OpenAPI method",
+                    f.item.elem.name()
+                ))
+            })
+            .collect::<Result<_>>()?;
+        paths.sort_by(|a, b| a.function_name.cmp(b.function_name));
```

Comment:
Thank you :)

Reply:
🥳 

I was having an issue that I’m surprised we hadn’t seen before, where the open API and typescript generators were producing slightly different list ordering for types, functions, classes, etc. It was only intermittently, though. I added some updates to the rust generators to sort the lists by name beforehand so it will be consistent every time. I haven’t updated Python or Ruby yet, though.